clientlibrary/worker/worker.go

```diff
@@ -100,8 +97,30 @@ type Worker struct {
 	mService      metrics.MonitoringService
 }
 
-// NewWorker constructs a Worker instance for processing Kinesis stream data.
 func NewWorker(factory kcl.IRecordProcessorFactory, kclConfig *config.KinesisClientLibConfiguration, metricsConfig *metrics.MonitoringConfiguration) *Worker {
+	w := newWorker(
```

Comment:
The worker should inject the checkpointer instead of creating it inside constructor.